docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -381,23 +408,46 @@ spec:
 
 * Infrastructure templates are expected to be immutable, so infrastructure template contents do not have to hashed in order to detect
   changes.
-   
+
+##### KubeadmControlPlane rollout (using delete-and-create)
+
+KCP rollout delete-and-create is recommended for resource constrained environment like bare-metal, OpenStack or vSphere resource pools, etc.
+
+- Triggered by:
+    - Changes to Version
+    - Changes to the kubeadmConfigSpec
+    - Changes to the infrastructureRef
+    - `KubeadmControlPlane.Spec.UpgradeStrategy` is set
```

Comment:
What if the value is set in the past?

Reply:
I did not understood what you mean?

Reply:
e.g What if `KubeadmControlPlane.Spec.UpgradeStrategy` is set to some time in the past, e.g. 1h ago? should scale in start in this case?

Reply:
I think it should. But if we want to have more control over upgrade maybe we should also use something similar to `upgradeAfter` here?

Reply:
Sorry, I made some confusion here. However:
- `KubeadmControlPlane.Spec.UpgradeStrategy` should be set to scale-in (not generically set)
- upgradeAfter should apply also to the scale-in strategy, but this will be implicitly defined by moving everything is common between the two strategies in the initial paragraph